src/v1/sources/shopify/transform.js

```diff
@@ -0,0 +1,314 @@
+  // attaching cart, checkout and order tokens in context object
+  message.setProperty(`context.cart_token`, event.cart_token);
+  message.setProperty(`context.checkout_token`, event.checkout_token);
+  if (shopifyTopic === 'orders_updated') {
+    message.setProperty(`context.order_token`, event.token);
+  }
+  message = removeUndefinedAndNullValues(message);
+  return message;
+};
+const isIdentifierEvent = (event) =>
+  ['rudderIdentifier', 'rudderSessionIdentifier'].includes(event?.event);
+const processIdentifierEvent = async (event, metricMetadata) => {
+  if (useRedisDatabase) {
+    const cartToken =
+      typeof event.cartToken === 'string' ? event.cartToken.split('?')[0] : event.cartToken;
+    logger.info(`{{SHOPIFY::}} writeKey: ${metricMetadata.writeKey}, cartToken: ${cartToken}`, {
+      type: 'set',
+      source: metricMetadata.source,
+      writeKey: metricMetadata.writeKey,
+    });
+    let value;
+    let field;
+    if (event.event === 'rudderIdentifier') {
+      field = 'anonymousId';
+      // eslint-disable-next-line unicorn/consistent-destructuring
+      const lineItemshash = getHashLineItems(event.cart);
+      // eslint-disable-next-line unicorn/consistent-destructuring
+      value = ['anonymousId', event.anonymousId, 'itemsHash', lineItemshash];
+      stats.increment('shopify_redis_calls', {
+        type: 'set',
+        field: 'itemsHash',
+        source: metricMetadata.source,
+        writeKey: metricMetadata.writeKey,
+      });
+      /* cart_token: {
+           anonymousId: 'anon_id1',
+           lineItemshash: '0943gh34pg'
+          }
+      */
+    } else {
+      field = 'sessionId';
+      // eslint-disable-next-line unicorn/consistent-destructuring
+      value = ['sessionId', event.sessionId];
+      /* cart_token: {
+          anonymousId:'anon_id1',
+          lineItemshash:'90fg348fg83497u',
+          sessionId: 'session_id1'
+         }
+       */
+    }
+    try {
+      stats.increment('shopify_redis_calls', {
+        type: 'set',
+        field,
+        source: metricMetadata.source,
+        writeKey: metricMetadata.writeKey,
+      });
+      await RedisDB.setVal(`${cartToken}`, value);
+    } catch (e) {
+      logger.debug(`{{SHOPIFY::}} cartToken map set call Failed due redis error ${e}`, {
+        type: 'set',
+        source: metricMetadata.source,
+        writeKey: metricMetadata.writeKey,
+      });
+      stats.increment('shopify_redis_failures', {
+        type: 'set',
+        source: metricMetadata.source,
+        writeKey: metricMetadata.writeKey,
+      });
+      // returning 500 as status code in case of redis failure
+      throw new RedisError(`${e}`, 500);
+    }
+  }
+  return NO_OPERATION_SUCCESS;
+};
+
+const process = async (inputEvent) => {
+  const { event, source } = inputEvent;
+  const metricMetadata = {
+    // eslint-disable-next-line unicorn/consistent-destructuring
+    writeKey: event.query_parameters?.writeKey?.[0],
```

Comment:
Check if we can get this from the source object

Reply:
Also lets add the `sourceId` label aswell

Reply:
Sure, we can get it from `source.WriteKey`.